src/http-api.js

```diff
@@ -74,12 +74,18 @@ module.exports.PREFIX_MEDIA_R0 = "/_matrix/media/r0";
  * requests.
  * @param {Number=} opts.localTimeoutMs The default maximum amount of time to wait
  * before timing out the request. If not specified, there is no timeout.
+ * @param {bool=} opts.useAuthorizationHeader True to use Authorization header instead of
+ * query param to send the access token to the server. Defaults to false
```

Comment:
JS SDK users do not create a `MatrixHttpApi`, so they will be unable to set this option without gut-wrenching `client._http`. This is set in the rather poorly defined "[base APIs](https://github.com/matrix-org/matrix-js-sdk/blob/598d40b0b75c50ef1c916b11c8dbac65af5f02ea/src/base-apis.js#L74)" constructor which is called in the [client constructor](https://github.com/matrix-org/matrix-js-sdk/blob/598d40b0b75c50ef1c916b11c8dbac65af5f02ea/src/client.js#L113).

Please bubble this down from `MatrixClient`, copying this documentation to the `opts` of `MatrixClient`.